### PR TITLE
Fix any behavior in SVE with garbage values

### PR DESCRIPTION
**Pull request by jfalcou**

### CODE REVIEW

File: include/eve/module/core/regular/impl/simd/arm/sve/any.hpp [hunk 1]
```diff
@@ -9,14 +9,17 @@
...
 
 namespace eve::detail
 {
 template<scalar_value T, typename N, relative_conditional_expr C>
 EVE_FORCEINLINE bool
-any_(EVE_SUPPORTS(sve_), C const& cond, logical<wide<T,N>> const& v) noexcept
+any_(EVE_SUPPORTS(sve_), C const& cond, logical<wide<T,N>> v) noexcept
 requires sve_abi<abi_t<T, N>>
 {
+  v = remove_garbage(v);
```
Comment (DenisYaroshevskiy) [inline include/eve/module/core/regular/impl/simd/arm/sve/any.hpp:21]: I don't like this very much.
Like for `svptest_any` we can put this remove garbage mask directly there.

Comment (jfalcou) [reply include/eve/module/core/regular/impl/simd/arm/sve/any.hpp]: Updated

File: include/eve/module/core/regular/impl/simd/arm/sve/any.hpp [hunk 2]
```diff
@@ -9,17 +9,22 @@
...
   {
-    if constexpr( C::is_inverted )  return svptest_any(sve_true<T>(), v);
+    if constexpr( C::is_inverted )
+    {
+      using ec_t  = expected_cardinal_t<T, typename logical<wide<T,N>>::abi_type>;
+      return svptest_any(keep_first(N::value).mask(as<wide<T,ec_t>>{}), v);
+    }
     else                            return false;
   }
   else return svptest_any(cond.mask(as<wide<T,N>>{}),v);
```
Comment (DenisYaroshevskiy) [inline include/eve/module/core/regular/impl/simd/arm/sve/any.hpp:30]: I think this should be

```
  // if not ignore all
      return svptest_any(sve_mask_for(cond, v), v);
```

or smth

Comment (jfalcou) [reply include/eve/module/core/regular/impl/simd/arm/sve/any.hpp:27]: turned into sve_true(v) as per discussion

**Review by DenisYaroshevskiy — APPROVED** [review]

File: include/eve/arch/arm/sve/sve_true.hpp [hunk 3]
```diff
@@ -24,4 +24,22 @@ sve_true()
...
+// while masking potential garbage value
+template<relative_conditional_expr C, typename T>
+EVE_FORCEINLINE svbool_t
+sve_true(C cond, as<T> tgt)
+{
+  if constexpr(C::is_complete && C::is_inverted)
+  {
+    using v_t   = element_type_t<T>;
+    using ec_t  = expected_cardinal_t<v_t, typename T::abi_type>;
+    return keep_first(T::size()).mask(as<as_wide_t<v_t,ec_t>>{});
```
Comment (DenisYaroshevskiy) [inline include/eve/arch/arm/sve/sve_true.hpp:38]: Ah - wait! Let's return sve_true() for just native register

File: include/eve/arch/arm/sve/sve_true.hpp [hunk 4]
```diff
@@ -24,4 +24,24 @@ sve_true()
...
+// while masking potential garbage value
+template<relative_conditional_expr C, typename T>
+EVE_FORCEINLINE svbool_t
+sve_true(C cond, as<T> tgt)
+{
+  if constexpr(C::is_complete && C::is_inverted)
+  {
+    using v_t   = element_type_t<T>;
+    using ec_t  = expected_cardinal_t<v_t, typename T::abi_type>;
+
```
Comment (DenisYaroshevskiy) [inline include/eve/arch/arm/sve/sve_true.hpp:38]: Ehy do you need abi here

**Review by DenisYaroshevskiy — APPROVED** [review]

File: include/eve/arch/arm/sve/sve_true.hpp [hunk 5]
```diff
@@ -24,4 +24,24 @@ sve_true()
...
+    if constexpr(ec_t::value == T::size())      return sve_true<v_t>();
```
Comment (DenisYaroshevskiy) [inline include/eve/arch/arm/sve/sve_true.hpp:39]: Hmm. I think this should actually be: `T::size() < native_cardinal_v<v_t>`

Comment (jfalcou) [reply include/eve/arch/arm/sve/sve_true.hpp:39]: no, if it is smaller, then we got garbage. Remeber than SVE size are one at a time, so haiving SVE512 has no SVE256.

File: include/eve/arch/arm/sve/sve_true.hpp (hunk 4 above)
Comment (jfalcou) [reply include/eve/arch/arm/sve/sve_true.hpp:38]: Turns out nope

**Pull request merged**